core/sdk-core/src/main/java/software/amazon/awssdk/core/interceptor/ExecutionAttribute.java

```diff
@@ -269,7 +269,10 @@ public void set(Map<ExecutionAttribute<?>, Object> attributes, T value) {
 
         @Override
         public void setIfAbsent(Map<ExecutionAttribute<?>, Object> attributes, T value) {
-            attributes.computeIfAbsent(realAttribute, k -> writeMapping.apply(null, value));
+            T currentValue = get(attributes);
+            if (currentValue == null) {
+                set(attributes, value);
+            }
```

Comment:
This is more correct, since the presence of the real attribute does not necessarily indicate that the execution attribute value is present (e.g. a signer property's existence isn't guaranteed even if the selected auth scheme is present).

Reply:
This fixed issues we've identified in other testing, even with new clients, since we still have code in core that does putIfAbsent after auth scheme is selected. Thank you!